runtime/vm/JVMImage.cpp

```diff
@@ -147,58 +151,70 @@ JVMImage::readImageFromFile()
 {
 	OMRPORT_ACCESS_FROM_OMRPORT(getPortLibrary());
 
-	// TODO: The size will need to be very large or dynamically based on whatever is written to the dump
-	// TODO: Fetch the initial size of the jvm image
-    char imageBuffer[JVMImage::INITIAL_IMAGE_SIZE];
-    memset(imageBuffer, 0, sizeof(imageBuffer));
+	omrthread_monitor_enter(_jvmImageMonitor);
```

Comment:
No need to thread lock for reading image. Also monitor is not created for warm runs (which uses reading)